pkgs/os-specific/linux/freeipa/default.nix

```diff
@@ -0,0 +1,97 @@
+{ stdenv, lib, fetchurl, writeText, pkgconfig, autoconf, automake, kerberos,
+  openldap, popt, sasl, curl, xmlrpc_c, ding-libs, p11-kit, gettext, nspr, nss,
+  _389-ds-base, svrcore, libuuid, talloc, tevent, samba, libunistring, libverto,
+  libpwquality, systemd, python3, bind, sssd, jre, rhino, lesscpy, jansson,
+  runtimeShell }:
+
+let
+  pathsPy = ./paths.py;
+
+  pythonInputs = with python3.pkgs; [
+    six ldap dns netaddr netifaces gssapi dogtag-pki pyasn1 sssd cffi lxml
+    dbus-python cryptography memcached qrcode pyusb yubico setuptools jinja2
+    augeas samba
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "freeipa";
+  version = "4.10.1";
+
+  src = fetchurl {
+    url = "https://releases.pagure.org/freeipa/freeipa-${version}.tar.gz";
+    sha256 = "sha256-q2rQzcBl1tI4/7+hxEwOY9ND86hObe7O7Y9EEH7cUoA=";
+  };
+
+  nativeBuildInputs = [
+    python3.pkgs.wrapPython jre rhino lesscpy automake autoconf gettext
+  ];
+
+  buildInputs = [
+    kerberos openldap popt sasl curl xmlrpc_c pkgconfig ding-libs p11-kit
+    python3 autoconf nspr nss _389-ds-base svrcore libuuid talloc tevent samba
+    libunistring libverto systemd bind libpwquality jansson
+  ] ++ pythonInputs;
+
+  postPatch = ''
+    patchShebangs makeapi makeaci install/ui/util
+
+    substituteInPlace ipaplatform/setup.py \
+      --replace 'ipaplatform.debian' 'ipaplatform.nixos'
```

Comment:
Personally I prefer that all things built from nixpkgs identify themselves as from nixpkgs and not from nixos. (There are already NixOS spin-offs.)

Reply:
This has to match what is pulled from uname. So it must be nixos...

Reply:
What if someone installs this on non-nixos with nix, like ubuntu?

Reply:
I suspect that nobody will want to use the nixos version of freeipa-client on ubuntu since there is a [native package available](https://packages.ubuntu.com/jammy/freeipa-client), but if that actually comes up we could make this overrideable in the derivation

Reply:
@Mindavi @benley Do I need to make changes?

Reply:
(IMHO it's fine as-is -- we can always improve later if we know of something better then.)

Reply:
So leaving it as is.